helm/flink-kubernetes-operator/conf/flink-conf.yaml

```diff
@@ -21,9 +21,11 @@ taskmanager.numberOfTaskSlots: 1
 parallelism.default: 1
 
 # These parameters are required for Java 17 support.
-# These should be kept in-sync with the flink dist env.java.opts.all defaults in: flink-dist/src/main/resources/config.yaml
-# default-opts.all is used here so users can supply their own opts.all in their Job deployments and have these appended.
-env.java.default-opts.all: --add-exports=java.base/sun.net.util=ALL-UNNAMED --add-exports=java.rmi/sun.rmi.registry=ALL-UNNAMED --add-exports=jdk.compiler/com.sun.tools.javac.api=ALL-UNNAMED --add-exports=jdk.compiler/com.sun.tools.javac.file=ALL-UNNAMED --add-exports=jdk.compiler/com.sun.tools.javac.parser=ALL-UNNAMED --add-exports=jdk.compiler/com.sun.tools.javac.tree=ALL-UNNAMED --add-exports=jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED --add-exports=java.security.jgss/sun.security.krb5=ALL-UNNAMED --add-opens=java.base/java.lang=ALL-UNNAMED --add-opens=java.base/java.net=ALL-UNNAMED --add-opens=java.base/java.io=ALL-UNNAMED --add-opens=java.base/java.nio=ALL-UNNAMED --add-opens=java.base/sun.nio.ch=ALL-UNNAMED --add-opens=java.base/java.lang.reflect=ALL-UNNAMED --add-opens=java.base/java.text=ALL-UNNAMED --add-opens=java.base/java.time=ALL-UNNAMED --add-opens=java.base/java.util=ALL-UNNAMED --add-opens=java.base/java.util.concurrent=ALL-UNNAMED --add-opens=java.base/java.util.concurrent.atomic=ALL-UNNAMED --add-opens=java.base/java.util.concurrent.locks=ALL-UNNAMED
```

Comment:
Why do we use version specific instead of general default? I think this should work even in older Flink versions

Reply:
Oh they will _currently_ work with older versions of Flink. The issue is that a future version of Flink might not need all these modules opened and/or exported and so that would break older version of Flink that do need them.

Reply:
I am looking at how we could get the Operator to just use the defaults from the Flink distribution directly so we don't need to do this.